nexus/reconfigurator/execution/src/lib.rs

```diff
@@ -316,6 +324,32 @@ fn register_deploy_disks_step<'a>(
         .register()
 }
 
+fn register_deploy_datasets_step<'a>(
+    registrar: &ComponentRegistrar<'_, 'a>,
+    opctx: &'a OpContext,
+    blueprint: &'a Blueprint,
+    sleds: SharedStepHandle<Arc<BTreeMap<SledUuid, Sled>>>,
+) {
```

Comment:
As a general feedback question, I've been idly wondering whether we should have a proc macro for steps, making them easier to define. Thoughts on that?

Reply:
I'm not sure yet -- each `register_...` function has a variable number of arguments, and the "call a function that calls `new_step`" pattern doesn't seem *terribly* onerous to me.

So, TL;DR, not *opposed* to a proc macro, just not sure it's necessary? But this is a weakly-held opinion

Reply:
I'd vote "no", personally, but recognize I might be in the minority. I find proc macros make some things more difficult (e.g., needing to answer "what will this expand to", and rust-analyzer's jump to only landing me in the crate that parses the proc macro), and those are the things I need when something is already going wrong; landing on a proc macro in that situation is annoying. I guess summarizing would be "they seem to make easy things easier but hard things harder", which doesn't seem worth it if the thing they're replacing isn't super onerous.